packages/evm/contracts/budgets/ManagedBudgetWithFees.sol

```diff
@@ -0,0 +1,272 @@
+
+        return true;
+    }
+
+    /// @inheritdoc AManagedBudgetWithFees
+    function payManagementFee(uint256 boostId, uint256 incentiveId) public override {
+        BoostLib.Boost memory boost = core.getBoost(boostId);
+
+        address validIncentive = address(boost.incentives[incentiveId]);
+
+        if (AIncentive(validIncentive).supportsInterface(type(AERC20Incentive).interfaceId)) {
+            uint256 claims = AERC20Incentive(validIncentive).claims();
+            uint256 limit = AERC20Incentive(validIncentive).limit();
+            uint256 balanceRemaining = AIncentive(validIncentive).asset().balanceOf(validIncentive);
+            if (claims == limit || balanceRemaining == 0) {
+                _transferManagementFee(boostId, incentiveId, boost, claims, limit);
+                return;
+            }
+            revert BoostError.FeePayoutFailed(balanceRemaining);
+        }
+        if (AIncentive(validIncentive).supportsInterface(type(AERC20VariableIncentive).interfaceId)) {
+            uint256 balanceRemaining = AIncentive(validIncentive).asset().balanceOf(validIncentive);
+            uint256 totalClaimed = AERC20VariableIncentive(validIncentive).totalClaimed();
+            uint256 limit = AERC20VariableIncentive(validIncentive).limit();
+            if (totalClaimed == limit || balanceRemaining == 0) {
+                _transferManagementFee(boostId, incentiveId, boost, totalClaimed, limit);
+                return;
+            }
+            revert BoostError.FeePayoutFailed(balanceRemaining);
+        }
+        revert BoostError.NotImplemented();
+    }
+
+    /// @inheritdoc AManagedBudgetWithFees
+    function clawbackFromTargetAndPayFee(address target, bytes calldata data_, uint256 boostId, uint256 incentiveId)
+        external
+        virtual
+        override
+        returns (uint256 amount, address asset)
+    {
+        (amount, asset) = clawbackFromTarget(target, data_, boostId, incentiveId);
+        payManagementFee(boostId, incentiveId);
+    }
+
+    /// @notice Transfers the management fee for a specific boost and incentive
+    /// @param boostId The ID of the boost for which the management fee is being transferred
+    /// @param incentiveId The ID of the incentive within the boost
+    /// @param boost The Boost object containing details about the boost
+    /// @param claim The number of claims made for the incentive
+    /// @param limit The limit of claims allowed for the incentive
+    /// @dev The function calculates the management fee based on the claims and
+    /// limit, then transfers the fee to the boost owner
+    function _transferManagementFee(
+        uint256 boostId,
+        uint256 incentiveId,
+        BoostLib.Boost memory boost,
+        uint256 claim,
+        uint256 limit
+    ) internal {
+        address incentive = address(boost.incentives[incentiveId]);
+        uint256 maxAmount = incentiveFees[incentive];
+        uint256 amount = maxAmount * claim / limit;
```

Comment:
I'm not sure this is the fee structure we want for a variable incentive - since limit is the total spend this would make the management fees pretty low for variable incentives.

Reply:
As we discussed, `claim` and `limit` can either be claim counts, or token amounts, depending on which type of ERC20 incentive is utilized:

https://github.com/boostxyz/boost-protocol/blob/2b9480e2df7f2ce3113896ab54fbf917bf15833c/packages/evm/contracts/budgets/ManagedBudgetWithFees.sol#L170

https://github.com/boostxyz/boost-protocol/blob/2b9480e2df7f2ce3113896ab54fbf917bf15833c/packages/evm/contracts/budgets/ManagedBudgetWithFees.sol#L180